part-1.py

```diff
@@ -4,17 +4,33 @@
 # the appropriate comment.
 
 # factorial
+def factorial(n):
+    if n < 0:
+        raise ValueError("Number must be greater than 0")
+    if n == 0:
+        return 1
 
-
+    return n * factorial(n - 1)
 
 # reverse
+def reverse(text):
```

Comment:
👍 Just a note that `text[:-1]` makes a copy of the array element-by-element, so that makes this recursive solution O(n^2)